src/v2/Components/UserSettings/__tests__/UserSettingsAddresses.jest.tsx

```diff
@@ -0,0 +1,92 @@
+import React from "react"
+import { UserSettingsAddressesFragmentContainer } from "../UserSettingsAddresses"
+import { setupTestWrapper } from "v2/DevTools/setupTestWrapper"
+import { graphql } from "relay-runtime"
+import ToastComponent from "v2/Components/Toast/ToastComponent"
+import { deleteUserAddress } from "v2/Apps/Order/Mutations/DeleteUserAddress"
+jest.unmock("react-relay")
+jest.mock("v2/Utils/Hooks/useMatchMedia", () => ({
+  useMatchMedia: () => ({}),
+}))
+jest.mock("v2/Apps/Order/Mutations/DeleteUserAddress")
+
+describe("ToastComponent", () => {
+  const { getWrapper } = setupTestWrapper({
+    Component: (props: any) => {
+      return <UserSettingsAddressesFragmentContainer me={props.me} />
+    },
+    query: graphql`
+      query UserSettingsAddresses_test_Query {
+        me {
+          ...UserSettingsAddresses_me
+        }
+      }
+    `,
+  })
+  const mockDeleteUserAddress = deleteUserAddress as jest.Mock
+  const wrapper = getWrapper({
+    Me: () => ({
+      id: "someid",
+      internalID: "someid",
+      addressConnection: mockAddressConnection,
+    }),
+  })
+
+  it("renders ToastComponent when address was saved", async () => {
+    const toast = wrapper.find(ToastComponent)
+    expect(toast.props().showNotification).toBe(false)
+
+    const deleteButton = wrapper
+      .find(`[data-test="deleteAddressInProfile"]`)
+      .first()
+    mockDeleteUserAddress.mockResolvedValue(mockDeleteAddressSuccessResponse)
+    deleteButton.simulate("click")
+    await wrapper.update()
+    const toast2 = wrapper.find(ToastComponent)
+    // I run here onShowToast func, but I can't get updating tree
+    //I tried to use setTimeout, but it doesn't work
+    setTimeout(() => {
+      console.log(toast2.text())
+      expect(toast2.props().showNotification).toBe(true)
```

Comment:
I know we talked a bit about tests for these and I think seeing as the `showNotification` prop is what is used to determine whether toast is visible this covers the functionality of the toast notif.

It might be worth adding tests for edit and save.

Reply:
I would be glad to create tests for save, but we call createAdressMutation inside AddressModal by clicking save(onSubmit for formik). I open this modal by simulating  click on edit but I can't get any components inside AddressModal after that to trigger createAdressMutation. Of course, I can achieve it by trigger inside AddressModal wrapper, but by ToastComponent is outside this wrapper, namely 2 level above